tools/integration_tests/managed_folders/admin_permissions_test.go

```diff
@@ -33,18 +35,163 @@ import (
+
+	err := os.RemoveAll(dirPath)
+	if err != nil {
+		t.Errorf("Error in removing managed folder: %v", err)
+	}
+}
+
+func (s *managedFoldersAdminPermission) TestCopyObjectWithInManagedFolder(t *testing.T) {
+	testDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1)
+	srcCopyFile := path.Join(testDirPath, FileInNonEmptyManagedFoldersTest)
+
+	destCopyFile := path.Join(testDirPath, DestFile)
+
+	err := operations.CopyFile(srcCopyFile, destCopyFile)
+	if err != nil {
+		t.Errorf("Error in copying file managed folder from src: %s to dest %s: %v", srcCopyFile, destCopyFile, err)
+	}
+
+	_, err = operations.StatFile(destCopyFile)
+	if err != nil {
+		t.Errorf("Error in stating destination copy file: %v", err)
+	}
+}
+
+func (s *managedFoldersAdminPermission) TestCopyManagedFolder(t *testing.T) {
+	if s.bucketPermission == ViewPermission {
+		// Copying the managed folder will fail because the bucket has view-only permissions.
+		return
+	}
+
+	srcDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1)
+	destDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, DestFolder)
+
+	err := operations.CopyDir(srcDirPath, destDirPath)
+	if err != nil {
+		t.Errorf("Error in copying directory: %v", err)
+	}
+
+	_, err = os.Stat(destDirPath)
+	if err != nil {
+		t.Errorf("Error in stating destination copy dir: %v", err)
+	}
+}
+
+func (s *managedFoldersAdminPermission) TestMoveObjectWithInManagedFolder(t *testing.T) {
+	testDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1)
+	srcMoveFile := path.Join(testDirPath, FileInNonEmptyManagedFoldersTest)
+
+	destMoveFile := path.Join(testDirPath, DestFile)
+
+	err := operations.Move(srcMoveFile, destMoveFile)
+	if err != nil {
+		t.Errorf("Error in moving file managed folder from src: %s to dest %s: %v", srcMoveFile, destMoveFile, err)
+	}
+
+	_, err = operations.StatFile(destMoveFile)
+	if err != nil {
+		t.Errorf("Error in stating destination move file: %v", err)
+	}
+	_, err = operations.StatFile(srcMoveFile)
+	if err == nil {
+		t.Errorf("SrcFile is not removed after move.")
+	}
+}
+
+func (s *managedFoldersAdminPermission) TestMoveManagedFolder(t *testing.T) {
+	if s.bucketPermission == ViewPermission {
+		// Moving the managed folder will fail because the bucket has view-only permissions.
+		return
+	}
+
+	srcDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1)
+	destDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, DestFolder)
+
+	err := operations.Move(srcDirPath, destDirPath)
+	if err != nil {
+		t.Errorf("Error in moving directory: %v", err)
+	}
+
+	_, err = os.Stat(destDirPath)
+	if err != nil {
+		t.Errorf("Error in stating destination copy dir: %v", err)
+	}
+	_, err = os.Stat(srcDirPath)
+	if err == nil {
+		t.Errorf("SrcDir is not removed after move.")
+	}
+}
+
+func (s *managedFoldersAdminPermission) TestMoveManagedFolderWithViewBucketPermission(t *testing.T) {
```

Comment:
Nit: I would suggest create a common test TestMoveManagedFolder and you can create 2 helper methods - validateMoveManagedFolderWithViewBucketPermission and validateManagedFolderWithAdminBucketPermission

Based on the type of permissions, you can do different validations. Setup and operation will be common for both.

Similar for TestCopyManagedFolder also.

Reply:
Good Idea. To more optimize it I have create single validator only with permission as argument.